### PR TITLE
nwg-look: init at v0.2.4

### LINKED ISSUE #246315: Package request: nwg-look
**Project description**
A GTK3 settings editor adapted to work in the wlroots environment, a part of the nwg-shell project.

**Metadata**

* homepage URL: https://nwg-piotr.github.io/nwg-shell/
* source URL: https://github.com/nwg-piotr/nwg-look
* license: MIT
* platforms: Linux

### PR DESCRIPTION
**Pull request by max-amb**
## Description of changes
https://github.com/nwg-piotr/nwg-look
## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

fixes #246315

### CODE REVIEW

File: pkgs/applications/misc/nwg-look/default.nix
Comment (khaneliman) [inline pkgs/applications/misc/nwg-look/default.nix:67]:
```suggestion
    maintainers = with maintainers; [ max-amb ];
    mainProgram = "nwg-look";
```

File: pkgs/applications/misc/nwg-look/default.nix
Comment (khaneliman) [inline pkgs/applications/misc/nwg-look/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchgit
, buildGoModule
, cairo
, glib
, go
, gtk3
, libX11
, pkg-config
, substituteAll
, xcur2png
, zlib
```

File: pkgs/applications/misc/nwg-look/default.nix
Comment (khaneliman) [inline pkgs/applications/misc/nwg-look/default.nix:41]:
```suggestion
    cairo
    glib
    gtk3
    libX11.dev
    xcur2png
    zlib
```

**Review by khaneliman — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/nwg-look/default.nix
Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:1]: Why, is it just for convention?

**Comment by max-amb** [comment]
Can I add a change to the maintainers list here as well?

File: pkgs/applications/misc/nwg-look/default.nix
Comment (khaneliman) [reply pkgs/applications/misc/nwg-look/default.nix]: just something ive been required to do before merges on all my PRs

**Comment by h4rldev** [comment]
When will this be merged? Any sort of ETA?

**Comment by max-amb** [comment]
Are there any issues you would like me to fix or can this be merged?

**Comment by h4rldev** [comment]
this seems mergable, I was the one that requested this to be ported.

**Comment by max-amb** [comment]
@khaneliman is there an issue you want me to fix?

**Comment by h4rldev** [comment]
Waiting for this to be merged still :+1:

**Comment by max-amb** [comment]
@drupol. Sorry to ping you but I was just wondering if there are any outstanding issues with this PR or can it be merged, sorry again for pinging you.

File: pkgs/applications/misc/nwg-look/default.nix
Comment (drupol) [inline pkgs/applications/misc/nwg-look/default.nix:28]: Each patch in nixpkgs applied to the upstream source should be documented. Out-of-tree patches, fetched using `fetchpatch`, are preferred.

In each patch comment, please explain the purpose of the patch and link to the relevant upstream issue if possible. If the patch has been merged upstream but is not yet part of the released version, please note the version number or date in the comment such that a future maintainer updating the nix expression will know whether the patch has been incorporated upstream and can thus be removed from nixpkgs.

Furthermore, please use a _stable_ URL for the patch. Rather than, for example, linking to a GitHub pull request of the form `https://github.com/owner/repo/pull/pr_number.patch`, which would change every time a commit is added or the PR is force-pushed, link to a specific commit patch in the form `https://github.com/owner/repo/commit/sha.patch`.

Here are two good examples of patch comments:

```nix
mkDerivation {
  …

  patches = [
    # Ensure RStudio compiles against R 4.0.0.
    # Should be removed next 1.2.X RStudio update or possibly 1.3.X.
    (fetchpatch {
      url = "https://github.com/rstudio/rstudio/commit/3fb2397c2f208bb8ace0bbaf269481ccb96b5b20.patch";
      sha256 = "0qpgjy6aash0fc0xbns42cwpj3nsw49nkbzwyq8az01xwg81g0f3";
    })
  ];
}
```

```nix
mkDerivation {
  …

  patches = [
    # Allow building with bison 3.7
    # PR at https://github.com/GoldenCheetah/GoldenCheetah/pull/3590
    (fetchpatch {
      url = "https://github.com/GoldenCheetah/GoldenCheetah/commit/e1f42f8b3340eb4695ad73be764332e75b7bce90.patch";
      sha256 = "1h0y9vfji5jngqcpzxna5nnawxs77i1lrj44w8a72j0ah0sznivb";
    })
  ];
}
```

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

File: pkgs/applications/misc/nwg-look/default.nix
Comment (andresilva) [reply pkgs/applications/misc/nwg-look/default.nix:28]: Did you look at the patch? It's specific to NixOS and even uses the same name as countless other patches in `nixpkgs` that do the exact same thing.

Comment (drupol) [reply pkgs/applications/misc/nwg-look/default.nix:28]: No, I haven't but I saw the patch filename. Just a simple sentence explaining the patch and that it is specific to nixos only is enough.

Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:31]: I have not used fetchpatch due to the patch being quite simple and it is not currently hosted on github, I think the comment details it well enough but let me know!

File: pkgs/applications/misc/nwg-look/default.nix
Comment (drupol) [inline pkgs/applications/misc/nwg-look/default.nix:24]: Prefer `hash = ...` to `sha256 = ...` Both currently work, but the former will make it easier if the algorithm ever needs to change.

Since there could be different hashing algorithms in use, this make sense.

File: pkgs/applications/misc/nwg-look/default.nix
Comment (drupol) [inline pkgs/applications/misc/nwg-look/default.nix:32]: 👍👍👍

File: pkgs/applications/misc/nwg-look/default.nix
Comment (drupol) [inline pkgs/applications/misc/nwg-look/default.nix:34]: How about using substituteInPlace ? Check in nixpkgs for plenty of examples

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/nwg-look/default.nix
Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:34]: I have had a look at using substituteInPlace and I have been unable to get it to work, I can try again but I doubt it would be succesful

File: pkgs/applications/misc/nwg-look/default.nix
Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:24]: Ok!

File: pkgs/applications/misc/nwg-look/default.nix
Comment (drupol) [reply pkgs/applications/misc/nwg-look/default.nix:34]: How it's going?

Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:34]: I still havent been able to get it to work using substituteInPlace

Comment (drupol) [reply pkgs/applications/misc/nwg-look/default.nix:34]: This doesn't work ?
```
  substituteInPlace main.go --replace '@out@' $out
```

File: pkgs/applications/misc/nwg-look/default.nix
Comment (drupol) [inline pkgs/applications/misc/nwg-look/default.nix:34]:
```suggestion
    substituteInPlace main.go tools.go --replace '@out@' $out
```

File: pkgs/applications/misc/nwg-look/default.nix
Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:34]: Im just testing!

File: pkgs/applications/misc/nwg-look/default.nix
Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:34]: Yes, seems to work, thanks!!

**Review by drupol — APPROVED** [review]

File: pkgs/applications/misc/nwg-look/default.nix
Comment (drupol) [inline pkgs/applications/misc/nwg-look/default.nix:22]: Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

Comment (drupol) [reply pkgs/applications/misc/nwg-look/default.nix:22]: Looks like I missed that in my review :(

**Comment by h4rldev** [comment]
When will this appear in the package database?

**Comment by LudovicoPiero** [comment]
You can check it [here](https://nixpk.gs/pr-tracker.html?pr=247157)

File: pkgs/applications/misc/nwg-look/default.nix
Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:22]: Do you want me to create a pr for this change or is it not that much of an issue?

Comment (drupol) [reply pkgs/applications/misc/nwg-look/default.nix:22]: If you're OK to do it, yes please, and ping me when it's done, I'll merge it straight.

Comment (max-amb) [reply pkgs/applications/misc/nwg-look/default.nix:22]: Ok, will do!

**Pull request merged**
